Sources/_SubprocessCShims/process_shims.c

```diff
@@ -664,6 +555,15 @@ int _subprocess_fork_exec(
         // If we reached this point, something went wrong
         write_error_and_exit;
     } else {
+        // On Linux 5.3 and lower, we have to fetch pidfd separately
+        // Newer Linux supports clone3 which returns pidfd directly
+        if (_linux_kernel_version_at_least(5, 3) == 0) {
+            _pidfd = _pidfd_open(childPid);
```

Comment:
as above, suggest just calling it without the version check and reacting accordingly if it returns ENOSYS